ukm-semantics/targets/execution/configuration.md

```diff
@@ -0,0 +1,35 @@
+```k
+
+requires "../../main/configuration.md"
+requires "../../main/preprocessed-configuration.md"
+requires "rust-semantics/config.md"
+
+module COMMON-K-CELL
+    imports private BYTES-SYNTAX
+    imports private INT-SYNTAX
+    imports private UKM-DECODING-SYNTAX
+    imports private UKM-EXECUTION-SYNTAX
+
+    configuration
+        <k>
+            ukmDecodePreprocessedCell($PGM:Bytes)
```

Comment:
What do we envision placing into the `ukmDecodePreprocessedCell`? Is it the hooks from the UKM module?

Reply:
The argument for the `ukmDecodePreprocessedCell` will be the bytes produced by `ukmEncodePreprocessedCell`.

`ukmDecodePreprocessedCell` will rewrite (in one or more steps) to `.K`. While doing so, it will "restore" the `<ukm-full-preprocessed>` cell (https://github.com/Pi-Squared-Inc/rust-demo-semantics/pull/143/files#diff-ab754f8b0d12c81607fb49790ffed6b962aeea3393f07b73f475aa5acc1f2e9aR28) to the state encoded in the `Bytes` argument.

This state will contain the normal content of the `<preprocessed>` cell in the Rust semantics (https://github.com/Pi-Squared-Inc/rust-demo-semantics/pull/143/files#diff-a93e9e66bc324710e7b57700a6758c826c8b7b704cda1ec9ae9e86c495ff0622R15) which contains, roughly, the "compiled" Rust code (we don't actually compile the code right now, we only do some sort of indexing, but, if the future we will probably do more, e.g. we will resolve imported symbols and so on). This state will also contain some contract-specific state (the `<ukm-preprocessed>` cell), e.g. it will hold a TypePath for the contract's trait, perhaps some other things, too.